src/test/java/net/datafaker/base/PhoneNumberTest.java

```diff
@@ -126,6 +126,486 @@ void testAllCellPhone_enPh() throws NumberParseException {
         assertThat(util.isValidNumberForRegion(proto, "PH")).as(phoneNumber).isTrue();
     }
 
+    @Test
```

Comment:
Why couldn't all these tests be just replaced with parameterized tests?
They are all the same except for locale... so it should be just a parameterized test with different inputs

Reply:
You're right, I will change it to use parameterized tests.

Reply:
And another question, what error count threshold will be good?

Reply:
what is error count?
from my point of view all the tests should be green

Reply:
> Region in new Locale("en", "IND") is different from region in util.parse(phoneNumber, "IN"). How to handle this case. 

There could be several input args: 2, 3, 4, ... whatever

Reply:
This is my test right now:
```
@ParameterizedTest
    @ValueSource(strings = {"US", "GB", "AU", "CA", "MS", "NG", "NZ"})
    void testPhoneNumberNational_en_language(String localeLanguage) throws NumberParseException {
        final PhoneNumberUtil util = PhoneNumberUtil.getInstance();
        int errorCount = 0;
        final BaseFaker faker = new BaseFaker(new Locale("en", localeLanguage));
        for (int i = 0; i < 10; i++) {
            String phoneNumber = faker.phoneNumber().phoneNumber();
            Phonenumber.PhoneNumber proto = util.parse(phoneNumber, localeLanguage);
            if (!util.isValidNumberForRegion(proto, localeLanguage)) {
                errorCount++;
            }
        }
        System.out.println(errorCount);
        assertThat(errorCount).isLessThan(250);
    }
```

Reply:
use `@MethodSource` it is more powerful
you can have a look at `net.datafaker.fileformats.JsonTest#simpleJsonTest` as an example

Reply:
Yes, it's really more powerful and seems to me that `@MethodSource` will work. Thank you

Reply:
@snuyanzin I didn't come up with something other than just adapting your code to this test